backend/azure/file.go

```diff
@@ -139,6 +139,9 @@ func (f *File) CopyToLocation(location vfs.Location) (vfs.File, error) {
 
 // CopyToFile puts the contents of the receiver (f *File) into the passed vfs.File parameter.
 func (f *File) CopyToFile(file vfs.File) error {
+	// Close file (f) reader regardless of an error
+	defer f.Close()
```

Comment:
`defer func() { _ = f.Close() }()`

Reply:
@funkyshu I thought about this but is there any reason that we would want to return the error from the Close() method?

Reply:
defer can't return an error, directly.  Close (on defer) is the most ignored error in go.  The only way you can really return an error in defer is by using a named error, but then you have protect against over writing a more important error.  Like lets say you're reading from a file and file doesn't exist.  That's the important thing to know.  Not that you couldn't also close the file, in defer.
So you have to do something like this: which may or may not be worth the time.  Having said that, the most important func in VFS is Close() because that's the single place where data is committed to the fileystem (on writes). So it MAY be worth ti.
```go
func (f *File) CopyToFile(file vfs.File) (err error) {
defer func() { 
    closeErr := f.Close()
    // only capture close err if there was no other error
    if closeErr != nil && err == nil {
          err = closeErr
    }
}()
```

An alternative is to concatenate multiple errors. 
```go
func (f *File) CopyToFile(file vfs.File) (err error) {
defer func() { 
    closeErr := f.Close()
    // only capture close err if there was no other error
    if closeErr != nil {
        if err == nil {
              err = closeErr
        } else {
              err = fmt.Errorf("orig error: %w - close err: %w", err, closeErr)
        } 
    }
}()

Reply:
Spoke with @funkyshu over slack and we cam eup with https://github.com/C2FO/vfs/pull/168/commits/1c2cce19f35944ce5cfaabe14dcddca3241905d3